frontend/src/components/widgets/inputs/FontInput.vue

```diff
@@ -125,50 +151,95 @@ export default defineComponent({
 			const selectedChoice = this.isStyle ? this.fontStyle : this.fontFamily;
 
-			let selectedEntry: MenuListEntry | undefined;
-			const menuListEntries = choices.map((name) => {
-				const result: MenuListEntry = {
-					label: name,
-					action: async (): Promise<void> => this.selectFont(name),
-				};
-
-				if (name === selectedChoice) selectedEntry = result;
-
-				return result;
-			});
-
-			const entries: SectionsOfMenuListEntries = [menuListEntries];
-			const activeEntry = selectedEntry || { label: "-" };
-
-			return { entries, activeEntry };
+			return entries.find((entry) => entry.value === selectedChoice) as MenuListEntry;
+		},
+		/// Handles keyboard navigation for the menu. Returns if the entire menu stack should be dismissed
+		keydown(e: KeyboardEvent) {
+			if (this.disabled) return;
+
+			const menuOpen = this.open;
+
+			if (!menuOpen && (e.key === " " || e.key === "Enter")) {
+				// Allow opening menu with space or enter
+				this.setOpen();
+			} else if (menuOpen && (e.key === "ArrowUp" || e.key === "ArrowDown")) {
+				// Navigate to the next and previous entries with arrow keys
+
+				let newIndex = e.key === "ArrowUp" ? this.entries.length - 1 : 0;
+				if (this.highlighted) {
+					const index = this.highlighted ? this.entries.indexOf(this.highlighted) : 0;
+					newIndex = (index + (e.key === "ArrowUp" ? -1 : 1) + this.entries.length) % this.entries.length;
+				}
+
+				const newEntry = this.entries[newIndex];
+				this.setHighlighted(newEntry);
+			} else if (menuOpen && e.key === "Escape") {
+				// Close menu with escape key
+				this.open = false;
+
+				// Reset active to before open
+				this.setHighlighted(this.activeEntry);
+			} else if (menuOpen && this.highlighted && e.key === "Enter") {
+				// Handle clicking on an option if enter is pressed
+				if (this.highlighted.value) this.selectFont(this.highlighted.value);
+				e.preventDefault();
+			}
+		},
+		setHighlighted(newHighlight: MenuListEntry | undefined) {
+			this.highlighted = newHighlight;
+		},
+		onScroll(e: Event) {
+			this.entriesStart = (e.target as HTMLElement)?.scrollTop || 0;
 		},
 	},
 	watch: {
 		async fontFamily() {
-			const { entries, activeEntry } = await this.updateEntries();
-			this.entries = entries;
-			this.activeEntry = activeEntry;
+			this.entries = await this.getEntries();
+			this.activeEntry = this.getActiveEntry(this.entries);
+			this.highlighted = this.activeEntry;
 		},
 		async fontStyle() {
-			const { entries, activeEntry } = await this.updateEntries();
-			this.entries = entries;
-			this.activeEntry = activeEntry;
+			this.entries = await this.getEntries();
+			this.activeEntry = this.getActiveEntry(this.entries);
+			this.highlighted = this.activeEntry;
+		},
+	},
+	computed: {
+		totalHeight() {
+			return this.entries.length * 20;
+		},
+		startIndex() {
+			return Math.floor(this.entriesStart / 20);
+		},
+		endIndex() {
+			return Math.min(this.entries.length, this.startIndex + 1 + 400 / 20);
```

Comment:
I think a lot of the code in here is now unused since it was moved to `MenuList.vue`?